tests/auth_mock.py

```diff
@@ -0,0 +1,331 @@
+                 {"url": endpoint, "interface": "admin", "region": None, "region_id": None, "id": ""}],
+              "type": "identity",
+              "id": "",
+              "name": "admin"}],
+         "extras": {},
+         'roles': [{'id': 'admin', 'name': 'admin'}],
+         "user": {"domain": {"id": "", "name": "admin"},
+                  "id": "abc", "name": "admin"},
+         "audit_ids": ["-SfXoGA"],
+         "issued_at": "2015-09-01T19:09:40.916497Z"}}
+
+
+headers = {
+    'Content-Type': 'application/json',
+}
+
+
+@httmock.urlmatch(path=r'.*/user-projects/$', method='GET')
+def get_user_projects(url, req):
+    auth_token = req.headers.get('X-Auth-Token')
+    if auth_token == Config.ADMIN_TOKEN:
+        return httmock.response(401)
+    elif auth_token == Config.FAIL_TOKEN:
+        return httmock.response(500)
+    else:
+        projects = Config.PROJECTS.get(auth_token)
+
+        if auth_token in Config.MEMBERS:
+            roles = ['_member_']
+        elif auth_token in Config.DOWNLOADERS:
+            roles = ['_member_', 'download']
+        elif auth_token in Config.SUBMITTERS:
+            roles = Config.ROLES
+        else:
+            return httmock.response(403)
+        return httmock.response(
+            200, json.dumps({project:roles for project in projects}),
+            headers = {'Content-Type': 'application/json'})
+
+
+@httmock.urlmatch(path=r'.*/auth/tokens$', method='GET')
+def get_auth_tokens(url, req):
+
+    # TODO populate with other portions of response
+
+    auth_token = req.headers.get('X-Auth-Token')
+    if auth_token != Config.ADMIN_TOKEN:
+        return httmock.response(401)
+
+    subject_token = req.headers.get('X-Subject-Token')
+    if subject_token == Config.FAIL_TOKEN:
+        return httmock.response(500)
+
+    headers = {
+        'Content-Type': 'application/json',
+        'X-Subject-Token': subject_token,
+    }
+
+    if subject_token in Config.VALID_TOKENS:
+        valid_token_resp = {
+            "token": {"methods": ["saml2"],
+                      "expires_at": "2015-10-01T18:04:56.623912Z",
+                      "extras": {},
+                      "user": {"OS-FEDERATION":
+                                {"identity_provider": {"id": "era_common"},
+                                 "protocol": {"id": "saml2"},
+                                 "groups": [{"id": subject_token}]},
+                               "id": subject_token,
+                               "name": subject_token},
+                      "audit_ids": ["abc"],
+                      "issued_at": "2015-09-01T18:04:56.624021Z"}}
+
+        return httmock.response(200, json.dumps(valid_token_resp), headers)
+    elif subject_token == Config.ADMIN_TOKEN:
+        return httmock.response(200, json.dumps(admin_token_resp), headers)
+    else:
+        return httmock.response(404)
+
+
+@httmock.urlmatch(path=r'.*/auth/tokens$', method='POST')
```

Comment:
What are these `'.*/auth/token$'` mocks for?

Reply:
Pulled from gdcapi. Not really understand, will check